### DISCUSSION TITLE
GitHub Checks Run Annotations - MarkDown Support

### DISCUSSION THREAD

**Discussion by jragula-zs** [community/community#58647] [Apps, API and Webhooks]
### Select Topic Area

General

### Body

We need the possibility of posting Markdown (MD) content within the output.annotations.message field of the API endpoint mentioned below:

API Endpoint: GitHub Checks Runs - Update a Check Run
https://docs.github.com/en/rest/checks/runs?apiVersion=2022-11-28#update-a-check-run

We have specific use case where we need to publish third-party check results on GitHub Pull Requests (PRs) utilizing the aforementioned API. To enhance the readability and formatting of the information presented, we are interested in leveraging Markdown capabilities.

The API right now doesn't reject posting MD content. But the content doesn't look formatted on Github UI.
So we need a feature to enhance the UI to display MD content for annotations messages.

**Comment by diclophis** [comment]
This feature would be a huge win for github actions checks UX.

**Comment by viluon** [comment]
Came here to request just this. The UI for check annotations seems very well suited to full Markdown messages. This would let us include more structured information, e.g. full stack traces for errors with clickable links to the other locations. Currently, annotations cannot have more than one location associated with them. Markdown messages would make that less of a problem.

![Screenshot from 2023-10-12 18-15-52](https://github.com/community/community/assets/7235381/bf6201ed-edec-4a7b-9a85-f98318ec7107)

@jragula-zs I think this might better fit the Actions category?
